File: app/src/test/java/com/example/adiphong/rx2/Concat.java
package com.example.adiphong.rx2;

import org.junit.Test;

import io.reactivex.Observable;

public class Concat {

    @Test
    public void concatCreate() {
        int input = 0;
        Observable.create(emitter -> {
                    if (!emitter.isDisposed()) {
                        int result = 1;
//                        result /= input;

                        emitter.onNext(result);
                        emitter.onComplete();
                    }
                }
        );

//        Observable<String> observable2 = Observable.create(emitter -> {
//            if (!emitter.isDisposed()) {
//                int result = 2;
////                result /= input;
//                emitter.onNext(""+result);
//                emitter.onComplete();
//            }
//        });
//
//        Observable.concat(observable1, observable2)
//                .doOnError(throwable -> System.out.println("--------- doOnError"))
//                .onErrorResumeNext(Observable.never())
//                .subscribe(
//                        s -> System.out.println("-------------- on next "+s),
//                        throwable -> System.out.println("----------- error "+throwable),
//                        () -> System.out.println("----------- on complete")
//                );
    }
}

File: app/src/test/java/com/example/adiphong/rx2/Concat.kt
package com.example.adiphong.rx2

import io.reactivex.Observable
import org.junit.Test

class Concat {

    @Test
    fun concatCreate() {
        val input = 0
        val observable1 = Observable.create<Int> { emitter ->
            if (!emitter.isDisposed) {
                var result = 1
//                 result /= input
                emitter.onNext(result)
                emitter.onComplete()
            }
        }

        val observable2 = Observable.create<String> { emitter ->
            if (!emitter.isDisposed) {
                var result = 2
//                                result /= input;
                emitter.onNext("" + result)
                emitter.onComplete()
            }
        }

        Observable.concat(observable1, observable2)
                .doOnError { println("--------- doOnError") }
                .onErrorResumeNext(Observable.never())
                .subscribe(
                        { s -> println("-------------- on next $s") },
                        { throwable -> println("----------- error $throwable") },
                        { println("----------- on complete") }
                )
    }
}
